src/internal/m365/graph/middleware.go

```diff
@@ -252,93 +245,88 @@ func (mw RetryMiddleware) retryRequest(
 	exponentialBackoff *backoff.ExponentialBackOff,
 	priorErr error,
 ) (*http.Response, error) {
-	status := "unknown_resp_status"
-	statusCode := -1
+	ctx = clues.Add(ctx, "retry_count", executionCount)
 
 	if resp != nil {
-		status = resp.Status
-		statusCode = resp.StatusCode
+		ctx = clues.Add(ctx, "prev_resp_status", resp.Status)
 	}
 
-	ctx = clues.Add(
-		ctx,
-		"prev_resp_status", status,
-		"retry_count", executionCount)
-
-	// only retry under certain conditions:
-	// 1, there was an error.  2, the resp and/or status code match retriable conditions.
+	// only retry if all the following conditions are met:
+	// 1, there was a prior error.
```

Comment:
I'm a little confused by this and the lead comment on the previous line. It seems like we want "there was a prior error or the below"

Reply:
ah, good catch, will fix